vortex-array/src/compute/slice.rs

```diff
@@ -26,12 +24,8 @@ pub fn slice(array: impl AsRef<ArrayData>, start: usize, stop: usize) -> VortexR
     let array = array.as_ref();
     check_slice_bounds(array, start, stop)?;
 
-    if array
-        .statistics()
-        .get_as(Stat::IsConstant)
-        .unwrap_or_default()
-    {
-        return Ok(ConstantArray::new(scalar_at(array, 0)?, stop - start).into_array());
+    if let Some(rhs) = array.constant() {
```

Comment:
```suggestion
    if let Some(const_scalar) = array.constant() {
```

`rhs` is an awkward name for this